src/Mod/Path/PathScripts/PathUtils.py

```diff
@@ -718,7 +722,30 @@ def __get_depths(self, equalstep=False):
             )[1:]
 
         depths.reverse()
-        return depths
+
+        if len(depths) < 2:
+            return depths
+
+        return self.__filter_roughly_equal_depths(depths)
+
+    def __filter_roughly_equal_depths(self, depths):
+        """Depths arrive sorted from largest to smallest, positive to negative.
+        Return unique list of depths, using PathGeom.isRoughly() method to determine
+        if the two values are equal.  Only one of two consecutive equals are removed.
+
+        The assumption is that there are not enough consecutively roughly-equal depths
+        to be removed, so as to eliminate an effective step-down depth with the removal
+        of repetitive roughly-equal values."""
+
+        # Do not need a copy of original `depths` because we are only reading and copying
+        depths.sort()  # Sort again in case user_depths were not sorted
+        keep = [depths[0]]  # initialize keep list with first element
+        for depth in depths[1:]:  # iterate the list from second element on
+            # Compare each to the last kept item
+            if not PathGeom.isRoughly(depth, keep[-1]):
+                keep.append(depth)  # if different, add to keep
+        keep.reverse()  # reverse results back high to low
+        return keep
```

Comment:
sliptonic's comment got me thinking on how to concisely do this. Note that I'm mostly writing this down so I'll remember it, it's not meant as an invitation for refactoring
```
def __filter_roughly_equal_depths(depths):
  last = None
  def unique(depths):
    for depth in sorted(depths):
      if last is None or not PathGeom.isRoughly(depth, last):
        last = depth
        yield i
  return list(unique(depths))
```

Reply:
Clever.  I've not used sorted().  It does the copy and sorting in one step without modifying the original. Nice.

@Russ4262 Feel free to kill the overly detailed comments.  That was just for our discussion.